typing/typecore.ml

```diff
@@ -734,21 +749,28 @@ let enter_variable ?(is_module=false) ?(is_as_variable=false) loc name mode ty
   if List.exists (fun {pv_id; _} -> Ident.name pv_id = name.txt)
       !pattern_variables
   then raise(Error(loc, Env.empty, Multiply_bound_variable name.txt));
-  let id = Ident.create_local name.txt in
+  let id =
+    if is_module then begin
+      (* Note: unpack patterns enter a variable of the same name *)
```

Comment:
The comment is transplanted. It could benefit from a reword --- how's this? This is how I would answer your questions based on the new wording:
  * the module variable/value variable have the same name as each other
  * yes, this is different than non-unpack patterns, where no module declaration is entered